### ISSUE TITLE
Unable to build on macOS Sierra (10.12.6) - Linkage to boost libraries seems to be broken.

### ISSUE THREAD

**Issue by chopson** [Microsoft/cpprestsdk#536]
Installed cpprestsdk, following instructions at https://github.com/Microsoft/cpprestsdk/wiki/How-to-build-for-Mac-OS-X 
	cpprestsdk	2.9.1
	cmake 		3.9.1
	boost 		1.65.0
	libiconv	        1.15
	openssl		1.0.21

> cmake ../Release -DCMAKE_BUILD_TYPE=Debug
-- Setting clang options
-- websocketpp not found, using the embedded version
CMake Warning at /usr/local/Cellar/cmake/3.9.1/share/cmake/Modules/FindBoost.cmake:771 (message):
  Imported targets not available for Boost version 106500
Call Stack (most recent call first):
  /usr/local/Cellar/cmake/3.9.1/share/cmake/Modules/FindBoost.cmake:875 (_Boost_COMPONENT_DEPENDENCIES)
  /usr/local/Cellar/cmake/3.9.1/share/cmake/Modules/FindBoost.cmake:1503 (_Boost_MISSING_DEPENDENCIES)
  cmake/cpprest_find_boost.cmake:21 (find_package)
  cmake/cpprest_find_websocketpp.cmake:15 (cpprest_find_boost)
  src/CMakeLists.txt:231 (cpprest_find_websocketpp)
...
(Repeated 8 times - I assume once for each of the boost libraries listed below: random, system, etc.)
...

-- Boost version: 1.65.0
-- Found the following Boost libraries:
--   random
--   system
--   thread
--   filesystem
--   chrono
--   atomic
--   date_time
--   regex
-- Added test library httpclient_test
-- Added test library httplistener_test
-- Added test library json_test
-- Added test library pplx_test
-- Added test library streams_test
-- Added test library uri_test
-- Added test library utils_test
-- Added test library websocketclient_test
-- Configuring done
CMake Error at src/CMakeLists.txt:123 (add_library):
  Target "cpprest" links to target "Boost::random" but the target was not
  found.  Perhaps a find_package() call is missing for an IMPORTED target, or
  an ALIAS target is missing?
...
(Repeated 168 times - once for each boost library for each of the 21 targets)
...
-- Generating done
-- Build files have been written to: /Users/cHopson/Development/casablanca/build.debug

What am I missing?  Or, what do I need to change in order to get things to compile & link?

Thanks!

**Comment by PhilCMiller** [comment]
I ran into this problem as well. I notice that Boost was updated on August 21, 2017, which is almost certainly when the problem was introduced for both of us.

**Comment by chopson** [comment]
Have you found any fix / workaround?  I found one suggestion to downgrade boost to an older version.  I haven not tried it yet - hoping for a real fix.

**Comment by PhilCMiller** [comment]
Being new to Brewmaster, I didn't find a way retrieve a previous version of Boost. As it happens, my project doesn't immediately the REST interface, so I'm also hoping for a real fix. BTW, I did verify that the Boost issue is independent of my project by attempting to build CPPRESTSDK with the documented build procedure (on Mac, in my case).

**Comment by PhilCMiller** [comment]
...doesn't immediately REQUIRE... (missing word)

**Comment by chopson** [comment]
Looking at using cpprestsdk in an upcoming project.  Trying to sort out the pros and cons (other than it doesn't build right out of the box!).  Also looking at just using URLSessing, etc.  Any thoughts / insights?

**Comment by PhilCMiller** [comment]
One key consideration for ANY REST implementation is security and how you can leverage it. My app is an on-premise server connecting to another on-premise server. Some of my customers do not allow external internet connections; some security mechanisms depend on them. Security for most apps depends on both authentication and authorization, which may be handled best with an internal mechanism (e.g., directory services or a platform service like NTLM); authorization is kind of platform- and application-dependent. I guess, in summary: figure out how much security you need, and figure out how to implement it.

**Comment by Ian4101** [comment]
It seems the cmake package is looking for boost versions up to anything less than 1.65. The error points to the area in the relevant file: /usr/local/Cellar/cmake/3.9.1/share/cmake/Modules/FindBoost.cmake:771. Changing the condition to < 106600 has addressed the build error for me.

**Comment by PhilCMiller** [comment]
Thanks for the tip! Not a cmake expert myself, the tip is most helpful!

**Comment by chopson** [comment]
Yes, thanks for the info.  I was going to make this change, but did not want to have to fix the 'next' issue.  Since you (Ian4101) said it 'addressed the build error' I pressed on. But wait, there's more...  Yes, I changed the version test to get past the first set of warnings & compiler errors.  However, when I ran make -j 4 (per the instructions) I got link errors.
/casablanca/build.debug/CMakeCache.txt:321: ZLIB_LIBRARY_RELEASE:FILEPATH=/usr/local/lib/libz.dylib
is looking for the wrong zlib.  I don't know who built this one, but it is built for i386 while the linker is looking for x86_64 symbols.
ld: warning: ignoring file /usr/local/lib/libz.dylib, file was built for i386 which is not the architecture being linked (x86_64): /usr/local/lib/libz.dylib
Undefined symbols for architecture x86_64:

The macOS build of libz.dylib (/usr/lib) is for both. Yes, I can change the path (again), but I only wanted to do a quick test to see if we could use casablanca. Perhaps I did not follow the installation instructions carefully.  Anyway, this sort of stuff makes me wary of trying to use something in a production setting.

**Comment by mobileben** [comment]
@chopson I think your problem is the cmake version

https://stackoverflow.com/questions/42123509/cmake-finds-boost-but-the-imported-targets-not-available-for-boost-version
